File: app/src/main/java/com/example/architechturestartercode/data/movie/datasource/remote/MoviesNetworkResponse.java
package com.example.architechturestartercode.data.movie.datasource.remote;

import com.example.architechturestartercode.data.movie.model.Movie;

import java.util.List;

public interface MoviesNetworkResponse {
    void onSuccess(List<Movie> movies);
    void onError(String errorMessage);
    void serverError(String serverErrorMessage);
}

File: app/src/main/java/com/example/architechturestartercode/data/movie/datasource/remote/MoviesNetworkResponse.kt
package com.example.architechturestartercode.data.movie.datasource.remote

import com.example.architechturestartercode.data.movie.model.Movie

interface MoviesNetworkResponse {
    fun onSuccess(movies: List<Movie>)
    fun onFailure(errorMessage: String)
    fun serverError(errorMessage: String)
}

